# Изучение развития игровой индустрии с 2000 по 2013 год

- Автор: Савич Ксения
- Дата:06.04.2025

### Цели и задачи проекта
- изучить развитие игровой индустрии с 2000 по 2013 год
- категоризовать игры по оценкам пользователей и экспертов
- вывести топ-7 платформ по количеству игр, выпущенных в данном периоде

### Описание данных

Данные /datasets/new_games.csv содержат информацию о продажах игр разных жанров и платформ, а также пользовательские и экспертные оценки игр:
- `Name` — название игры.
- `Platform` — название платформы.
- `Year of Release` — год выпуска игры.
- `Genre` — жанр игры.
- `NA sales` — продажи в Северной Америке (в миллионах проданных копий).
- `EU sales` — продажи в Европе (в миллионах проданных копий).
- `JP sales` — продажи в Японии (в миллионах проданных копий).
- `Other sales` — продажи в других странах (в миллионах проданных копий).
- `Critic Score` — оценка критиков (от 0 до 100).
- `User Score` — оценка пользователей (от 0 до 10).
- `Rating` — рейтинг организации ESRB (англ. Entertainment Software Rating Board). Эта ассоциация определяет рейтинг компьютерных игр и присваивает им подходящую возрастную категорию.

### Содержимое проекта
1. Загрузка данных и знакомство с ними
2. Проверка ошибок в данных и их предобработка
3. Фильтрация данных
4. Категоризация данных
5. Выводы
---

## 1. Загрузка данных и знакомство с ними

- На данном этапе выгружаем необходимые библиотеки Python и данные датасета `/datasets/new_games.csv`. Затем изучаем данные при помощи методов `info()` и `head()`.


In [132]:
# Загружаем библиотеку pandas для дальнешей работы над данными и даем ей наименование "pd"
import pandas as pd

In [133]:
# Выгружаем данные датасета и даем ему наименование "df"
df = pd.read_csv('/datasets/new_games.csv')

In [134]:
#Знакомимся с данными при помощи метода info()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  object 
 1   Platform         16956 non-null  object 
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  object 
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  object 
 6   JP sales         16956 non-null  object 
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  object 
 10  Rating           10085 non-null  object 
dtypes: float64(4), object(7)
memory usage: 1.4+ MB


In [135]:
# Выводим первые строки данных при помощи метода head()
df.head()

Name Platform  Year of Release         Genre  NA sales  \
0                Wii Sports      Wii           2006.0        Sports     41.36   
1         Super Mario Bros.      NES           1985.0      Platform     29.08   
2            Mario Kart Wii      Wii           2008.0        Racing     15.68   
3         Wii Sports Resort      Wii           2009.0        Sports     15.61   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing     11.27   

  EU sales JP sales  Other sales  Critic Score User Score Rating  
0    28.96     3.77         8.45          76.0          8      E  
1     3.58     6.81         0.77           NaN        NaN    NaN  
2    12.76     3.79         3.29          82.0        8.3      E  
3    10.93     3.28         2.95          80.0          8      E  
4     8.89    10.22         1.00           NaN        NaN    NaN

Выводы после знакомства с данными:
- данные представляют собой 11 столбцов и 16956 строк
- типы данных float64 и object
- в столбцах `Name`, `Year of Release`, `Genre`, `Critic Score`, `User Score`, `Rating` есть пропуски(более чем в половине столбцов). В столбце `Critic Score` отсутсвует более половины данных
- названия в данных не приведены к языку snake case, что в дальнейшем следует исправить
- столбцы `EU sales`, `JP sales` и `User Score` следует привести к типу данных float64, так как они представляют собой числа и так с ними можно будет выполнять математичексие операции. Столбец `Year of Release` не выйдет привести к типу данных datetime64, т к данные представлены в виде года. Приведем столбцы `Year of Release` и `Critic Score` к типу данных int64.

---

## 2.  Проверка ошибок в данных и их предобработка


### 2.1. Названия, или метки, столбцов датафрейма

В данной части проекта приведем названия стобцов к snake case(заменим пробелы нижними подчеркиваниями и приведем все к нижнему регистру).

In [136]:
# Выводим названия всех столбцов
df.columns

Index(['Name', 'Platform', 'Year of Release', 'Genre', 'NA sales', 'EU sales',
       'JP sales', 'Other sales', 'Critic Score', 'User Score', 'Rating'],
      dtype='object')

In [137]:
# Приводим названия столбцов к snake case
df.columns = df.columns.str.lower().str.replace(' ', '_')

### 2.2. Типы данных

Преобразуем столбцы с неверным типом данных к верным.

In [138]:
# Приводим столбцы eu_sales, jp_sales и user_score к типу данных float64
df['eu_sales'] = pd.to_numeric(df['eu_sales'], errors='coerce')
df['jp_sales'] = pd.to_numeric(df['jp_sales'], errors='coerce')
df['user_score'] = pd.to_numeric(df['user_score'], errors='coerce')
df['year_of_release'] = pd.to_numeric(df['year_of_release'], errors='coerce')
df['critic_score'] = df['critic_score'].astype('Int64')
df['year_of_release'] = df['year_of_release'].astype('Int64')

In [139]:
# Проверяем, что типы данных изменились
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             16954 non-null  object 
 1   platform         16956 non-null  object 
 2   year_of_release  16681 non-null  Int64  
 3   genre            16954 non-null  object 
 4   na_sales         16956 non-null  float64
 5   eu_sales         16950 non-null  float64
 6   jp_sales         16952 non-null  float64
 7   other_sales      16956 non-null  float64
 8   critic_score     8242 non-null   Int64  
 9   user_score       7688 non-null   float64
 10  rating           10085 non-null  object 
dtypes: Int64(2), float64(5), object(4)
memory usage: 1.5+ MB


### 2.3. Наличие пропусков в данных

Считаем количество пропусков, анализируем их природу и очищаем данные от пропусков.


In [140]:
# Подсчитываем количество пропусков в каждом столбце
df.isna().sum()

name                  2
platform              0
year_of_release     275
genre                 2
na_sales              0
eu_sales              6
jp_sales              4
other_sales           0
critic_score       8714
user_score         9268
rating             6871
dtype: int64

In [141]:
# Количество пропусков в относительных значениях
df.isnull().mean()

name               0.000118
platform           0.000000
year_of_release    0.016218
genre              0.000118
na_sales           0.000000
eu_sales           0.000354
jp_sales           0.000236
other_sales        0.000000
critic_score       0.513918
user_score         0.546591
rating             0.405225
dtype: float64

Выводы о пропусках:
- Пропусков достаточно много, больше всего пропусков в столбцах с оценками, что легко объясняется тем, что конкретной игре могли просто не поставить оценок. Данные с оценками нужны и пропусков слишком много, поэтому пропуски необходимо заменить. Заменим значения индикатором, который не помешает дальнейшему анализу 101 и 11 соотвественно, т к у нас 100-балльная система и 10-балльная система в данных столбцах.
- Пропуски в столбцах с именем и жанром случайны и их меньше процента, также эти столбцы не важны для дальнеших операций, поэтому данные пропуски можно проигнорировать.
- Стобец с годом выхода важен для дальнейшего анализа, пропуски в нем составляют порядка 1,6%. В данном случае не выйдет заменить пропущенные знания, т к даты выяснить невозможно, а брать средние значения было бы некорректно. Пропуски в годе выхода необходимо удалить. 
- Пропуски с продажами в Японии и Европе можно заменить на среднее значение в зависимости от названия платформы и года выхода игры.
- Пропусков в столбце с рейтингом много - 40%. Заменим их индикатором "Unknown".


In [142]:
# Создадим копию нашего датасета, чтобы первоначальные данные остались в исходном виде
# Это необходимо для анализа объема упраздненных данных
df_copy = df.copy()

In [143]:
# Удалим данные с пропусками в столбце year_of_release
df_copy = df_copy.dropna(subset=['year_of_release'])

In [144]:
# Создаем функцию для заполнения пропусков и заменяем пропуски в продажах средним значением по платформе и году выхода
def fill_missing(group):
    return group.fillna(group.mean())

df_copy['eu_sales'] = df_copy.groupby(['platform', 'year_of_release'])['eu_sales'].apply(fill_missing)
df_copy['jp_sales'] = df_copy.groupby(['platform', 'year_of_release'])['jp_sales'].apply(fill_missing)


In [145]:
# Заменим пропуски в оценках критиков и пользователей на индикаторы
df_copy['user_score'] = df_copy['user_score'].fillna(11)
df_copy['critic_score'] = df_copy['critic_score'].fillna(101)

In [146]:
# Замним пропуски в столбце rating на индикатор "Unknown"
df_copy['rating'] = df_copy['rating'].fillna('Unknown')

In [147]:
df_copy.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16681 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             16679 non-null  object 
 1   platform         16681 non-null  object 
 2   year_of_release  16681 non-null  Int64  
 3   genre            16679 non-null  object 
 4   na_sales         16681 non-null  float64
 5   eu_sales         16681 non-null  float64
 6   jp_sales         16681 non-null  float64
 7   other_sales      16681 non-null  float64
 8   critic_score     16681 non-null  Int64  
 9   user_score       16681 non-null  float64
 10  rating           16681 non-null  object 
dtypes: Int64(2), float64(5), object(4)
memory usage: 1.6+ MB


In [148]:
df_copy.head()

name platform  year_of_release         genre  na_sales  \
0                Wii Sports      Wii             2006        Sports     41.36   
1         Super Mario Bros.      NES             1985      Platform     29.08   
2            Mario Kart Wii      Wii             2008        Racing     15.68   
3         Wii Sports Resort      Wii             2009        Sports     15.61   
4  Pokemon Red/Pokemon Blue       GB             1996  Role-Playing     11.27   

   eu_sales  jp_sales  other_sales  critic_score  user_score   rating  
0     28.96      3.77         8.45            76         8.0        E  
1      3.58      6.81         0.77           101        11.0  Unknown  
2     12.76      3.79         3.29            82         8.3        E  
3     10.93      3.28         2.95            80         8.0        E  
4      8.89     10.22         1.00           101        11.0  Unknown

Замена пропусков прошла успешно. Можно предположить, что оценок у этих игр действительно не было, т к при анализе первых строк, оказалось, что игры без оценок старые(например: Супер Марио 1985 года или Покемон 1996 года). Пропуски заменили на индикаторы, чтобы в дальнейшем анализе не было проблем в применении методов из-за пропусков.

Заменили пропуски на значения 11 и 101. Оценки 10-ти и 100 балльные, поэтому в дальнейшем пропуски не войдут в категории оценок. Для поиска среднего и т д нам эти данные не нужны, поэтому индикаторы подобраны удачно для наших целей.

### 2.4. Явные и неявные дубликаты в данных

Изучаем данные на предмет явных и неявных дубликатов, проводим очистку от дубликатов.

In [149]:
#Выявляем количество дубликатов
df_copy.duplicated().sum()

179

In [150]:
# Проверяем дубликаты в названиях жанра, платформы, рейтинга, года выпуска
print(df_copy['genre'].unique())
print(df_copy['platform'].unique())
print(df_copy['rating'].unique())
print(df_copy['year_of_release'].unique())

['Sports' 'Platform' 'Racing' 'Role-Playing' 'Puzzle' 'Misc' 'Shooter'
 'Simulation' 'Action' 'Fighting' 'Adventure' 'Strategy' nan 'MISC'
 'ROLE-PLAYING' 'RACING' 'ACTION' 'SHOOTER' 'FIGHTING' 'SPORTS' 'PLATFORM'
 'ADVENTURE' 'SIMULATION' 'PUZZLE' 'STRATEGY']
['Wii' 'NES' 'GB' 'DS' 'X360' 'PS3' 'PS2' 'SNES' 'GBA' 'PS4' '3DS' 'N64'
 'PS' 'XB' 'PC' '2600' 'PSP' 'XOne' 'WiiU' 'GC' 'GEN' 'DC' 'PSV' 'SAT'
 'SCD' 'WS' 'NG' 'TG16' '3DO' 'GG' 'PCFX']
['E' 'Unknown' 'M' 'T' 'E10+' 'K-A' 'AO' 'EC' 'RP']
<IntegerArray>
[2006, 1985, 2008, 2009, 1996, 1989, 1984, 2005, 1999, 2007, 2010, 2013, 2004,
 1990, 1988, 2002, 2001, 2011, 1998, 2015, 2012, 2014, 1992, 1997, 1993, 1994,
 1982, 2016, 2003, 1986, 2000, 1995, 1991, 1981, 1987, 1980, 1983]
Length: 37, dtype: Int64


In [151]:
df_copy.head()

name platform  year_of_release         genre  na_sales  \
0                Wii Sports      Wii             2006        Sports     41.36   
1         Super Mario Bros.      NES             1985      Platform     29.08   
2            Mario Kart Wii      Wii             2008        Racing     15.68   
3         Wii Sports Resort      Wii             2009        Sports     15.61   
4  Pokemon Red/Pokemon Blue       GB             1996  Role-Playing     11.27   

   eu_sales  jp_sales  other_sales  critic_score  user_score   rating  
0     28.96      3.77         8.45            76         8.0        E  
1      3.58      6.81         0.77           101        11.0  Unknown  
2     12.76      3.79         3.29            82         8.3        E  
3     10.93      3.28         2.95            80         8.0        E  
4      8.89     10.22         1.00           101        11.0  Unknown

В столбце с жанром есть дубликаты, связанные с регистром. В других проверяемых столбцах дубликатов нет.

In [152]:
# Меняем регистр столбца с жанрами для устранения дубликатов
df_copy['genre'] = df_copy['genre'].str.lower()

In [153]:
# Проверяем столбец на дубликаты после очистки
print(df_copy['genre'].unique())

['sports' 'platform' 'racing' 'role-playing' 'puzzle' 'misc' 'shooter'
 'simulation' 'action' 'fighting' 'adventure' 'strategy' nan]


In [154]:
# Смотрим, сколько осталось дубликатов в данных
df_copy.duplicated().sum()

235

Дубликатов стало больше, что говорит о наличии явных дубликатов

In [155]:
df_sorted = df_copy.sort_values(by=df_copy.columns.tolist())

# Находим дубликаты
duplicates = df_sorted[df_sorted.duplicated(keep=False)]
print(duplicates)

                                   name platform  year_of_release  \
15191                    Beyblade Burst      3DS             2016   
15192                    Beyblade Burst      3DS             2016   
15301                 11eyes: CrossOver     X360             2009   
15302                 11eyes: CrossOver     X360             2009   
4860   18 Wheeler: American Pro Trucker      PS2             2001   
...                                 ...      ...              ...   
2909   Yu-Gi-Oh! The Falsebound Kingdom       GC             2002   
6695                      Zoo Resort 3D      3DS             2011   
6696                      Zoo Resort 3D      3DS             2011   
8156                 Zumba Fitness Rush     X360             2012   
8157                 Zumba Fitness Rush     X360             2012   

              genre  na_sales  eu_sales  jp_sales  other_sales  critic_score  \
15191  role-playing      0.00      0.00      0.03         0.00           101   
15192  role

In [156]:
df_copy = df_copy.drop_duplicates()

Итого мы удалили ровно 235 явных дубликатов. Ничего лишнего!

В процессе обработки данных от дубликатов и пропусков, мы удаляли данные. Посмотрим, сколько данных мы удалили в абсолютном и относительном значениях.

In [157]:
# Подсчитаем количество удаленных данных в абсолютном и в относительном значениях
a = len(df) - len(df_copy)
b = a/len(df)*100
print(f'Убрали строк {a}, доля убранных строк в процентах {b}')

Убрали строк 510, доля убранных строк в процентах 3.007784854918613


В процессе обработки данных, было удалено 510 строк, что составляет 3% от всех данных. Думаю, это приемлемое количество. Удалены были дубликанты и те пропуски, которые нельзя учитывать в исследовании. Остальные пропуски были заменены подходящими значениями, либо оставлены без изменений.

---

## 3. Фильтрация данных

Отберем данные за 2000-2013 годы. Для этого создадим новый датафрейм `df_actual`

In [158]:
# Создадим новый датафрейм с данными за 2000-2013 год
start_date = 2000.0
end_date = 2013.0
df_actual = df_copy[(df_copy['year_of_release'] >= start_date) & (df_copy['year_of_release'] <= end_date)]

In [159]:
#Проверяем начало нового датафрейма
df_actual.head()

name platform  year_of_release     genre  na_sales  \
0             Wii Sports      Wii             2006    sports     41.36   
2         Mario Kart Wii      Wii             2008    racing     15.68   
3      Wii Sports Resort      Wii             2009    sports     15.61   
6  New Super Mario Bros.       DS             2006  platform     11.28   
7               Wii Play      Wii             2006      misc     13.96   

   eu_sales  jp_sales  other_sales  critic_score  user_score rating  
0     28.96      3.77         8.45            76         8.0      E  
2     12.76      3.79         3.29            82         8.3      E  
3     10.93      3.28         2.95            80         8.0      E  
6      9.14      6.50         2.88            89         8.5      E  
7      9.18      2.93         2.84            58         6.6      E

Данные успешно отобраны. Проверили начало нового датафрейма методом `head()`


## 4. Категоризация данных
    
Проведем категоризацию данных:
- Разделим все игры по оценкам пользователей и выделим такие категории: высокая оценка (от 8 до 10 включительно), средняя оценка (от 3 до 8, не включая правую границу интервала) и низкая оценка (от 0 до 3, не включая правую границу интервала).

In [160]:
# Для корректной работы следующего кода создаю копию нового датарейма, так как предыдущий был срезом
# данных и код работал некорректно
df_fin = df_actual.copy()

In [161]:
# Создаем столбец с категоризацией данных на основе оценок пользователей
bins = [0, 3, 8, 10, 12]  # Определяем границы категорий
labels = ['низкая оценка', 'средняя оценка', 'высокая оценка', 'нет оценки'] # Присваиваем категории
df_fin.loc[:, 'user_score_category'] = pd.cut(df_fin['user_score'], bins=bins, labels=labels, right=False)
# Благодаря right=False мы исключаем правую границу из интервала

- Разделим все игры по оценкам критиков и выделим такие категории: высокая оценка (от 80 до 100 включительно), средняя оценка (от 30 до 80, не включая правую границу интервала) и низкая оценка (от 0 до 30, не включая правую границу интервала).

In [162]:
# Создаем столбец с категоризацией данных на основе оценок критиков
bins = [0, 30, 80, 100, 102]  # Определяем границы категорий
labels = ['низкая оценка', 'средняя оценка', 'высокая оценка', 'нет оценки'] # Присваиваем категории
df_fin.loc[:, 'critic_score_category'] = pd.cut(df_fin['critic_score'], bins=bins, labels=labels, right=False)
# Благодаря right=False мы исключаем правую границу из интервала

In [165]:
df_fin.head()

name platform  year_of_release     genre  na_sales  \
0             Wii Sports      Wii             2006    sports     41.36   
2         Mario Kart Wii      Wii             2008    racing     15.68   
3      Wii Sports Resort      Wii             2009    sports     15.61   
6  New Super Mario Bros.       DS             2006  platform     11.28   
7               Wii Play      Wii             2006      misc     13.96   

   eu_sales  jp_sales  other_sales  critic_score  user_score rating  \
0     28.96      3.77         8.45            76         8.0      E   
2     12.76      3.79         3.29            82         8.3      E   
3     10.93      3.28         2.95            80         8.0      E   
6      9.14      6.50         2.88            89         8.5      E   
7      9.18      2.93         2.84            58         6.6      E   

  user_score_category critic_score_category  
0      высокая оценка        средняя оценка  
2      высокая оценка        высокая оценка  
3      высокая оценка        высокая оценка  
6      высокая оценка        высокая оценка  
7      средняя оценка        средняя оценка

In [164]:
df_fin.tail()

name platform  \
16947                     Men in Black II: Alien Escape       GC   
16949                Woody Woodpecker in Crazy Castle 5      GBA   
16950  SCORE International Baja 1000: The Official Game      PS2   
16952                                  LMA Manager 2007     X360   
16954                                  Spirits & Spells      GBA   

       year_of_release     genre  na_sales  eu_sales  jp_sales  other_sales  \
16947             2003   shooter      0.01      0.00       0.0          0.0   
16949             2002  platform      0.01      0.00       0.0          0.0   
16950             2008    racing      0.00      0.00       0.0          0.0   
16952             2006    sports      0.00      0.01       0.0          0.0   
16954             2003  platform      0.01      0.00       0.0          0.0   

       critic_score  user_score   rating user_score_category  \
16947           101        11.0        T          нет оценки   
16949           101        11.0  Unknown          нет оценки   
16950           101        11.0  Unknown          нет оценки   
16952           101        11.0  Unknown          нет оценки   
16954           101        11.0  Unknown          нет оценки   

      critic_score_category  
16947            нет оценки  
16949            нет оценки  
16950            нет оценки  
16952            нет оценки  
16954            нет оценки

Категории успешно присвоены

- После категоризации данных проверим результат: сгруппируем данные по выделенным категориям и посчитаем количество игр в каждой категории.

In [166]:
df_fin.groupby('user_score_category').count()

name  platform  year_of_release  genre  na_sales  \
user_score_category                                                     
низкая оценка         116       116              116    116       116   
средняя оценка       4081      4081             4081   4081      4081   
высокая оценка       2286      2286             2286   2286      2286   
нет оценки           6298      6298             6298   6298      6298   

                     eu_sales  jp_sales  other_sales  critic_score  \
user_score_category                                                  
низкая оценка             116       116          116           116   
средняя оценка           4081      4081         4081          4081   
высокая оценка           2286      2286         2286          2286   
нет оценки               6298      6298         6298          6298   

                     user_score  rating  critic_score_category  
user_score_category                                             
низкая оценка               116     116                    116  
средняя оценка             4081    4081                   4081  
высокая оценка             2286    2286                   2286  
нет оценки                 6298    6298                   6298

In [167]:
df_fin.groupby('critic_score_category').count()

name  platform  year_of_release  genre  na_sales  \
critic_score_category                                                     
низкая оценка            55        55               55     55        55   
средняя оценка         5422      5422             5422   5422      5422   
высокая оценка         1692      1692             1692   1692      1692   
нет оценки             5612      5612             5612   5612      5612   

                       eu_sales  jp_sales  other_sales  critic_score  \
critic_score_category                                                  
низкая оценка                55        55           55            55   
средняя оценка             5422      5422         5422          5422   
высокая оценка             1692      1692         1692          1692   
нет оценки                 5612      5612         5612          5612   

                       user_score  rating  user_score_category  
critic_score_category                                           
низкая оценка                  55      55                   55  
средняя оценка               5422    5422                 5422  
высокая оценка               1692    1692                 1692  
нет оценки                   5612    5612                 5612

- Выделим топ-7 платформ по количеству игр, выпущенных за весь актуальный период.

In [168]:
platform_counts = df['platform'].value_counts() # Подсчитываем количество игр в разрезе платформы
top_7_platforms = platform_counts.head(7) # Выводим топ-7 платформ
print(top_7_platforms)

PS2     2189
DS      2177
PS3     1355
Wii     1340
X360    1281
PSP     1229
PS      1215
Name: platform, dtype: int64


---

## 5. Итоговый вывод


- в ходе выполнения проекта была проведена предобработка данных посредством выявления, анализа и обработки пропусков и дубликатов. Данные были очищены от 510 строк, что составляет 3% данных. Также была проведена работа по замене типов данных, где это требовалось, и приведения названий столбцов к snake case. 
- далее была произведена фильтрация для отбора данных за необходимый период 2000-2013гг. Для этого понадобилось создать новый датафрейм `df_actual`, чтобы сохранить исходные данные в первоначальном виде.
- была проведена категоризация данных, для чего были созданы новые столбцы `user_score_category`, `critic_score_category`, были подсчитаны значения в разрезе данных категорий.
- был выделен топ-7 платформ по количеству игр, выпущенных за актуальный период.

В столбцах с оценками критиков и пользователей было много пропусков(почти половина данных), что может быть связано с ранним выходом игры. Почти для всех старых игр в данных столбцах значился пропуск. К сожалению, игры без оценок являются самой большой категорией.<br>
В ходе анализа можно заметить, что критики более склонны к усредненным оценкам, они менее категоричны, чем игроки. Игроки более склонны ставить высокий и низкий балл. Но в большинстве своем и критики и пользователи ставят играм среднюю оценку.
<br>
Также мы получили список 7 самых активных по продажам платформ: PS2, DS, PS3, Wii, X360, PSP, PS.    